source/guide_synapse.rst

```diff
@@ -0,0 +1,412 @@
+    Installing collected packages: pyasn1, pyasn1-modules, attrs, constantly, idna, hyperlink, zope.interface, six, Automat, PyHamcrest, incremental, Twisted, msgpack, simplejson, frozendict, canonicaljson, sortedcontainers, pyyaml, pycparser, cffi, pynacl, asn1crypto, cryptography, service-identity, unpaddedbase64, signedjson, pillow, daemonize, chardet, certifi, urllib3, requests, treq, bcrypt, pymacaroons, pyopenssl, psutil, pyrsistent, jsonschema, netaddr, phonenumbers, prometheus-client, matrix-synapse
+    Successfully installed Automat-0.7.0 PyHamcrest-1.9.0 Twisted-19.2.1 asn1crypto-0.24.0 attrs-19.1.0 bcrypt-3.1.6 canonicaljson-1.1.4 certifi-2019.6.16 cffi-1.12.3 chardet-3.0.4 constantly-15.1.0 cryptography-2.7 daemonize-2.5.0 frozendict-1.2 hyperlink-19.0.0 idna-2.8 incremental-17.5.0 jsonschema-3.0.1 matrix-synapse-1.0.0 msgpack-0.6.1 netaddr-0.7.19 phonenumbers-8.10.13 pillow-6.0.0 prometheus-client-0.3.1 psutil-5.6.3 pyasn1-0.4.5 pyasn1-modules-0.2.5 pycparser-2.19 pymacaroons-0.13.0 pynacl-1.3.0 pyopenssl-19.0.0 pyrsistent-0.15.2 pyyaml-5.1.1 requests-2.22.0 service-identity-18.1.0 signedjson-1.0.0 simplejson-3.16.0 six-1.12.0 sortedcontainers-2.1.0 treq-18.6.0 unpaddedbase64-1.1.0 urllib3-1.25.3 zope.interface-4.6.0
+  (env) [isabell@stardust ~]$ pip install jinja2
+    Collecting jinja2
+      Using cached https://files.pythonhosted.org/packages/1d/e7/fd8b501e7a6dfe492a433deb7b9d833d39ca74916fa8bc63dd1a4947a671/Jinja2-2.10.1-py2.py3-none-any.whl
+    Collecting MarkupSafe>=0.23 (from jinja2)
+      Using cached https://files.pythonhosted.org/packages/b2/5f/23e0023be6bb885d00ffbefad2942bc51a620328ee910f64abe5a8d18dd1/MarkupSafe-1.1.1-cp36-cp36m-manylinux1_x86_64.whl
+    Installing collected packages: MarkupSafe, jinja2
+    Successfully installed MarkupSafe-1.1.1 jinja2-2.10.1
+  (env) [isabell@stardust ~]$ pip install psycopg2
+    Collecting psycopg2
+      Downloading https://files.pythonhosted.org/packages/5c/1c/6997288da181277a0c29bc39a5f9143ff20b8c99f2a7d059cfb55163e165/psycopg2-2.8.3.tar.gz (377kB)
+         |████████████████████████████████| 378kB 21.4MB/s
+    Building wheels for collected packages: psycopg2
+      Building wheel for psycopg2 (setup.py) ... done
+      Stored in directory: /home/matrites/.cache/pip/wheels/48/06/67/475967017d99b988421b87bf7ee5fad0dad789dc349561786b
+    Successfully built psycopg2
+    Installing collected packages: psycopg2
+    Successfully installed psycopg2-2.8.3
+
+  (env) [isabell@stardust ~]$
+
+
+Configuration
+=============
+
+Generate standard config
+------------------------
+
+Generate a config file ``~/synapse/homeserver.yaml`` and replace my.domain.name with the real domain you want your matrix username to end in.
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ source ~/synapse/env/bin/activate
+  (env) [isabell@stardust ~]$ cd ~/synapse
+  (env) [isabell@stardust synapse]$  python -m synapse.app.homeserver \
```

Comment:
This needs `jinja2`, which is not listed as a dependency. Please install it manually during the installation step, for now. The [bugfix for this](https://github.com/matrix-org/synapse/pull/5514/files) is not yet released :/

```
[matrixx@helin synapse]$ python3.6 -m synapse.app.homeserver \
>   --server-name my.domain.name \
>   --config-path homeserver.yaml \
>   --generate-config \
>   --report-stats=yes
Traceback (most recent call last):
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/matrixx/.local/lib/python3.6/site-packages/synapse/app/homeserver.py", line 64, in <module>
    from synapse.rest import ClientRestResource
  File "/home/matrixx/.local/lib/python3.6/site-packages/synapse/rest/__init__.py", line 32, in <module>
    from synapse.rest.client.v2_alpha import (
  File "/home/matrixx/.local/lib/python3.6/site-packages/synapse/rest/client/v2_alpha/account.py", line 21, in <module>
    import jinja2
ModuleNotFoundError: No module named 'jinja2'
```

Reply:
thought I already added it